sdk/core/azure-core/inc/http/policy.hpp

```diff
@@ -117,4 +125,37 @@ namespace Azure { namespace Core { namespace Http {
     }
   };
 
+  class TelemetryPolicy : public HttpPolicy {
+    std::string m_telemetryId;
+
+    static std::string const g_emptyApplicationId;
+
+    static std::string BuildTelemetryId(
+        std::string const& componentName,
+        std::string const& componentVersion,
+        std::string const& applicationId);
+
+  public:
+    explicit TelemetryPolicy(std::string const& componentName, std::string const& componentVersion)
+        : TelemetryPolicy(componentName, componentVersion, g_emptyApplicationId)
+    {
+    }
+
+    explicit TelemetryPolicy(
```

Comment:
Can you make this work more like this: https://github.com/Azure/azure-storage-blob-go/blob/master/azblob/zc_policy_telemetry.go

Specifically, the ctor should take 1 options parameter which has 1 field which shoul dbe call application_id to match this guidelines: https://azure.github.io/azure-sdk/general_azurecore.html (the go code calls this value but the go code should change to application_id)

Then, the rest of the string should be created by the policy itself so that all callers don't have to redo the same work all the time and this also ensures that the string is correct and consistent when sent to the the service.

Reply:
Per our standup conversation, I am not doing this. The behavior is copied from .NET SDK, and the Go factory you mentioned does create the policy for a specific module, this can be done in C++, but should not be done at this (core) level.